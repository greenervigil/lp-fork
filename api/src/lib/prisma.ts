import { PrismaClient } from '@prisma/client';

const {
    POSTGRES_DATABASE,
    POSTGRES_HOST,
    POSTGRES_PASSWORD,
    POSTGRES_SCHEMA,
    POSTGRES_URL,
    POSTGRES_USER
} = process.env;

export const prisma = new PrismaClient({
    datasources: {
        db: {
            url: POSTGRES_URL
                ? POSTGRES_URL
                : `postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:5432/${POSTGRES_DATABASE}?schema=${POSTGRES_SCHEMA}"`
        }
    }
});
