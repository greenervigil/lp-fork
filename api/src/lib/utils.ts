import bcrypt from 'bcrypt';

const SALT_ROUNDS = 10;

const compareHash = async (password: string, hash: string) =>
    bcrypt.compare(password, hash);

const encryptHash = async (password: string) => {
    const salt = await bcrypt.genSalt(SALT_ROUNDS);

    return bcrypt.hash(password, salt);
};
