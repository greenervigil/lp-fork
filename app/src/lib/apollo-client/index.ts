import {
    ApolloClient,
    ApolloLink,
    HttpLink,
    Observable,
    from,
    fromPromise,
    split,
    toPromise
} from '@apollo/client';

import { ApolloError } from 'apollo-server-errors';
import { GraphQLWsLink } from '@apollo/client/link/subscriptions';
import { createClient } from 'graphql-ws';
import { getMainDefinition } from '@apollo/client/utilities';
import { getSession } from 'next-auth/react';
import initCache from './init-cache';
import { onError } from '@apollo/client/link/error';
import router from 'next/router';

export type ApolloClientOptions = {
    session: boolean;
    uri: string;
};

/**
 * Omega Apollo Client
 *
 * @param {ApolloClientOptions} ApolloClientOptions
 * @returns {ApolloClient}
 *
 * @see https://www.apollographql.com/docs/react/api/core/ApolloClient
 */
export const client = ({
    session,
    uri
}: ApolloClientOptions): ApolloClient<unknown> => {
    const ssrMode = typeof window === 'undefined';
    const cache = initCache({});
    const httpLink = new HttpLink({
        uri
    });

    let link = httpLink;

    if (!ssrMode) {
        const wsLink = new GraphQLWsLink(
            createClient({
                url: 'ws://localhost:4422/policies/graphql'
            })
        );

        link = split(
            ({ query }) => {
                const definition = getMainDefinition(query);

                return (
                    definition.kind === 'OperationDefinition' &&
                    definition.operation === 'subscription'
                );
            },
            wsLink,
            httpLink
        );
    }

    const redirectToLogin = () => {
        router.push({
            pathname: '/sign-in',
            query: {
                redirect: router.asPath
            }
        });
    };

    const sessionLink = new ApolloLink((operation, forward) => {
        const observable = forward(operation);

        return fromPromise(
            getSession().then(data => {
                return toPromise(
                    new Observable(observer => {
                        return observable.subscribe({
                            next: value => {
                                if (!data) {
                                    value.errors = [
                                        new ApolloError('AUTHENTICATION ERROR')
                                    ];
                                }
                                observer.next(value);
                            },
                            error: observer.error.bind(observer),
                            complete: observer.complete.bind(observer)
                        });
                    })
                );
            })
        );
    });

    const errorLink = onError(({ graphQLErrors }) => {
        if (graphQLErrors) {
            graphQLErrors.forEach(({ message }) => {
                if (/AUTHENTICATION ERROR/.test(message)) {
                    redirectToLogin();
                }
            });
        }
    });

    return new ApolloClient({
        link: from(
            session ? [errorLink, sessionLink, link] : [errorLink, link]
        ),
        cache,
        ssrMode,
        connectToDevTools: !ssrMode,
        name: 'omega-client'
    });
};
