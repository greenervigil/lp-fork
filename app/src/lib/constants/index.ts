export const REQUIRED_FIELD_MESSAGE = 'Required.';

export const DATE_FORMAT_PATTERN = 'MMM d, yyyy';
