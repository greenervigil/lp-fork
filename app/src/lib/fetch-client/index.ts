import axios from 'axios';

const { DATA_GRAPH_HOST } = process.env ?? {};

/**
 * Reusable Fetch based Apollo client query.
 *
 * @param {*} query
 * @param {*} options
 * @returns
 */
export const fetchQuery = async (query: any, options?: any) => {
    if (DATA_GRAPH_HOST) {
        const { headers, variables } = options ?? {};

        const { data } = await axios.post(
            DATA_GRAPH_HOST,
            JSON.stringify({
                query,
                variables
            }),
            {
                // timeout: 1000,
                withCredentials: true,
                headers: {
                    ...headers,
                    'Content-Type': 'application/json'
                }
            }
        );

        if (data.errors) {
            console.error(data.errors, 'fetchQuery');
        }

        return data;
    } else {
        throw new Error('DATA_GRAPH_HOST is missing.');
    }
};

// Used for IDE highlighting
export const gql = (str: any) => str.toString();
