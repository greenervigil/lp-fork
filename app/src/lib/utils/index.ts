export { formatDate, formatDateRelative } from './date';
export { formatFullName } from './string';
