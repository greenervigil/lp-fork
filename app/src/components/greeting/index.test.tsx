import '@testing-library/jest-dom';

import * as fixtures from '@/lib/fixtures';

import { render, screen } from '@testing-library/react';

import Greeting from '.';

describe('Greeting', () => {
    describe('component', () => {
        const student = fixtures.student();
        const now = new Date();
        now.setHours(9);

        it('should render', () => {
            const name = 'Frank';

            const { baseElement } = render(<Greeting name={name} date={now} />);
            expect(baseElement).toMatchSnapshot();
            expect(screen.getByTestId('greeting')).toBeInTheDocument();
        });

        it('should display name', () => {
            render(<Greeting name={student.firstName} date={now} />);

            expect(screen.getByTestId('greeting')).toHaveTextContent(
                student.firstName
            );
        });

        it('should display morning greeting', () => {
            now.setHours(8);

            render(<Greeting name={student.firstName} date={now} />);

            expect(screen.getByTestId('greeting')).toHaveTextContent(
                'Good Morning'
            );
        });

        it('should display afternoon greeting', () => {
            now.setHours(14);

            render(<Greeting name={student.firstName} date={now} />);

            expect(screen.getByTestId('greeting')).toHaveTextContent(
                'Good Afternoon'
            );
        });

        it('should display evening greeting', () => {
            now.setHours(18);

            render(<Greeting name={student.firstName} date={now} />);

            expect(screen.getByTestId('greeting')).toHaveTextContent(
                'Good Evening'
            );
        });
    });
});
