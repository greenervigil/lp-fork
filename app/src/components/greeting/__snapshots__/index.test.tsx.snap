// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting component should render 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
      data-testid="greeting"
    >
      Good Morning
      , 
      Frank
    </div>
  </div>
</body>
`;
